python/amici/parameter_mapping.py

```diff
@@ -0,0 +1,368 @@
+        amici_model: amici.Model) -> None:
+    """Fill fixed and dynamic parameters into the edata for condition
+    (in-place).
+
+    :param edata:
+        Experimental data object to fill parameters into.
+    :param problem_parameters:
+        Problem parameters as parameterId=>value dict. Only
+        parameters included here will be set. Remaining parameters will
+        be used as already set in `amici_model` and `edata`.
+    :param scaled_parameters:
+        If True, problem_parameters are assumed to be on the scale provided
+        in the parameter mapping. If False, they
+        are assumed to be in linear scale.
+    :param parameter_mapping:
+        Parameter mapping for current condition.
+    :param amici_model:
+        AMICI model
+    """
+    map_sim_var = parameter_mapping.map_sim_var
+    scale_map_sim_var = parameter_mapping.scale_map_sim_var
+    map_preeq_fix = parameter_mapping.map_preeq_fix
+    scale_map_preeq_fix = parameter_mapping.scale_map_preeq_fix
+    map_sim_fix = parameter_mapping.map_sim_fix
+    scale_map_sim_fix = parameter_mapping.scale_map_sim_fix
+
+    # Parameter mapping may contain parameter_ids as values, these *must*
+    # be replaced
+
+    def _get_par(model_par, value):
+        """Replace parameter IDs in mapping dicts by values from
+        problem_parameters where necessary"""
+        if isinstance(value, str):
+            # estimated parameter
+            # (condition table overrides must have been handled already,
+            # e.g. by the PEtab parameter mapping)
+            return problem_parameters[value]
+        if model_par in problem_parameters:
+            # user-provided
+            return problem_parameters[model_par]
+        # constant value
+        return value
+
+    map_preeq_fix = {key: _get_par(key, val)
+                     for key, val in map_preeq_fix.items()}
+    map_sim_fix = {key: _get_par(key, val)
+                   for key, val in map_sim_fix.items()}
+    map_sim_var = {key: _get_par(key, val)
+                   for key, val in map_sim_var.items()}
+
+    # If necessary, (un)scale parameters
+    if scaled_parameters:
+        unscale_parameters_dict(map_preeq_fix, scale_map_preeq_fix)
+        unscale_parameters_dict(map_sim_fix, scale_map_sim_fix)
+    if not scaled_parameters:
+        # We scale all parameters to the scale they are estimated on, and pass
+        # that information to amici via edata.{parameters,pscale}.
+        # The scaling is necessary to obtain correct derivatives.
+        scale_parameters_dict(map_sim_var, scale_map_sim_var)
+        # We can skip preequilibration parameters, because they are identical
+        # with simulation parameters, and only the latter are used from here
+        # on.
+
+    ##########################################################################
+    # variable parameters and parameter scale
+
+    # parameter list from mapping dict
+    parameters = [map_sim_var[par_id]
```

Comment:
this will always be a list, so the if below will always yield true. Check for `len(map_sim_var)` here and set parameters to `None` (or better directly combine with setting of parameters in edata). Same goes with scale.